src/p5.dance.interpreted.js

```diff
@@ -93,38 +134,60 @@ function ifDanceIs(sprite, dance, ifStatement, elseStatement) {
   }
 }
 
-function whenKey(key, event) {
+/**
+ * @param {string} key - Many options, including up, down, left, right, a-z, 0-9
+ * @param {function} func - Code to run when event fires
+ */
+function whenKey(key, func) {
   inputEvents.push({
     type: 'this.p5_.keyWentDown',
-    event: event,
-    param: key
+    func: func,
+    param: key,
+    queueType: QueueType.other,
   });
 }
 
-function whenPeak(range, event) {
+/**
+ * @param {string} range - Should be "bass", "mid", or "treble"
+ * @param {function} func - Code to run when event fires
+ */
+function whenPeak(range, func) {
   inputEvents.push({
     type: 'Dance.fft.isPeak',
-    event: event,
-    param: range
+    func: func,
+    param: range,
+    queueType: QueueType.other,
   });
 }
 
-function atTimestamp(timestamp, unit, event) {
+/**
+ * @param {number} timestamp
+ * @param {string} unit - Should be "measures" or "seconds"
+ * @param {function} func - Code to run when event fires
+ */
+function atTimestamp(timestamp, unit, func) {
+  // Despite the functions name, if we call atTimestamp(4, "measures", foo), we
```

Comment:
Well, we shall see about this separately.

Reply:
Not blocking this PR, I hope!